mshadow/cuda/tensor_gpu-inl.cuh

```diff
@@ -6,6 +6,11 @@
  */
 #ifndef MSHADOW_CUDA_TENSOR_GPU_INL_CUH_
 #define MSHADOW_CUDA_TENSOR_GPU_INL_CUH_
+#include <thrust/device_ptr.h>
```

Comment:
Is thrust installed with cuda by default? Does this complicate compiling and linking? Do we need to update config and cmake?